EDU-12464 feat: add Displaying SLA by item in Checkout (#1258)
vtexdocs/dev-portal-content: docs/guides/Checkout/customization/displaying-sla-by-item-in-checkout.md

---
title: "Displaying SLA by item in Checkout"
slug: "displaying-sla-by-item-in-checkout"
hidden: false
createdAt: ""
updatedAt: ""
excerpt: "This guide describes how to display individual shipping estimates for items in a customer's cart on an ecommerce platform using the Checkout API."
hidePaginationPrevious: false
hidePaginationNext: false
---

To enhance the customer experience, some stores may want to display precise shipping estimates for each item in the customer's cart. This can be particularly useful for ecommerce platforms that have sellers with different shipping capabilities and delivery times. This guide will walk through the process of displaying the Service Level Agreement (SLA) by item using the [Checkout API](https://developers.vtex.com/docs/api-reference/checkout-api).

## Enable the `useIndividualShippingEstimates` flag

> First, it is necessary to enable the [Checkout API](https://developers.vtex.com/docs/api-reference/checkout-api) behavior to fill the `shippingEstimateDate` with the SLA information. To do so, you must update the store's `orderForm` with the [Update orderForm configuration](https://developers.vtex.com/docs/api-reference/checkout-api#post-/api/checkout/pvt/configuration/orderForm) endpoint. You must include the flag `useIndividualShippingEstimates = true` as the example below:

Reviewer: ```suggestion
First, it is necessary to enable the [Checkout API](https://developers.vtex.com/docs/api-reference/checkout-api) behavior to fill the `shippingEstimateDate` with the SLA information. To do so, update the store's `orderForm` using the [Update orderForm configuration](https://developers.vtex.com/docs/api-reference/checkout-api#post-/api/checkout/pvt/configuration/orderForm) endpoint and set the flag `useIndividualShippingEstimates` to `true` as shown in the example below:
```

```json
{
    "paymentConfiguration": {
    ...
    },
    "taxConfiguration": null,
    "minimumQuantityAccumulatedForItems": 1,
    "decimalDigitsPrecision": 2,
    "minimumValueAccumulated": null,
    "apps": [
    ...
    ],
    "allowMultipleDeliveries": true,
    "allowManualPrice": null,
    "savePersonalDataAsOptIn": false,
    "maxNumberOfWhiteLabelSellers": null,
    "recaptchaValidation": "vtexcriteria",
    "recaptchaMinScore": null,
    "recaptchaKeys": null,
    "maskStateOnAddress": true,
    "enableSecureCookies": true,
    "useOwnershipCookie": true,
    "ignoreProfileData": null,
    "useIndividualShippingEstimates": true
}
```

> By doing so, the `shippingEstimateDate` will be updated with the SLA information configured by the sellers.

Reviewer: ```suggestion
By enabling this flag, the `shippingEstimateDate` will be updated with the SLA information configured by the sellers.
```

## Add items to cart

> Once you enable the `useIndividualShippingEstimates`, you can test if the configuration is correct by adding items to a shopping cart using the [Add cart items](https://developers.vtex.com/docs/api-reference/checkout-api#post-/api/checkout/pub/orderForm/-orderFormId-/items) endpoint. See the request body example below:

Reviewer: ```suggestion
Once you enable the `useIndividualShippingEstimates`, test if the configuration is correct by adding items to a shopping cart using the [Add cart items](https://developers.vtex.com/docs/api-reference/checkout-api#post-/api/checkout/pub/orderForm/-orderFormId-/items) endpoint. See the request body example below:
```

```json
{
  "orderItems": [
    {
      "seller": "1",
      "id": 6,
      "quantity": 1
    },
    {
      "seller": "1",
      "id": 11,
      "quantity": 1
    }
  ]
}
```

## Send shipping data attachment

> Send the item shipping data to the `orderForm` by using the [Add shipping address and select delivery option](https://developers.vtex.com/docs/api-reference/checkout-api#post-/api/checkout/pub/orderForm/-orderFormId-/attachments/shippingData) endpoint as the example below:

Reviewer: ```suggestion
Send the item shipping data to the `orderForm` using the [Add shipping address and select delivery option](https://developers.vtex.com/docs/api-reference/checkout-api#post-/api/checkout/pub/orderForm/-orderFormId-/attachments/shippingData) endpoint as the example below:
```

```json
{
  "selectedAddresses": [
    {
      "addressId": "5139683010323",
      "addressType": "search",
      "city": null,
      "complement": null,
      "country": "BRA",
      "neighborhood": null,
      "number": null,
      "postalCode": "08391705",
      "receiverName": "Jane Doe",
      "reference": null,
      "street": null,
      "addressQuery": "",
      "isDisposable": null
    }
  ],
  "expectedOrderFormSections": [
    "items",
    "totalizers",
    "clientProfileData",
    "shippingData",
    "paymentData",
    "sellers",
    "messages",
    "marketingData",
    "clientPreferencesData",
    "storePreferencesData",
    "giftRegistryData",
    "ratesAndBenefitsData",
    "openTextField",
    "commercialConditionData",
    "customData"
  ]
}
```

> Once configured, the `orderForm` will retrieve the variable `shippingEstimateDate` with the shipping estimate date being displayed, instead of only the shipping estimate in business days.

Reviewer: ```suggestion
Once configured, the `orderForm` will retrieve the variable `shippingEstimateDate`, which will display the shipping estimate date instead of only the shipping estimate in business days.
```

---
title: "Displaying SLA by item in Checkout"
slug: "displaying-sla-by-item-in-checkout"
hidden: false
createdAt: ""
updatedAt: ""
excerpt: "This guide describes how to display individual shipping estimates for items in a customer's cart on an ecommerce platform using the Checkout API."
hidePaginationPrevious: false
hidePaginationNext: false
---

To enhance the customer experience, some stores may want to display precise shipping estimates for each item in the customer's cart. This can be particularly useful for ecommerce platforms that have sellers with different shipping capabilities and delivery times. This guide will walk through the process of displaying the Service Level Agreement (SLA) by item using the [Checkout API](https://developers.vtex.com/docs/api-reference/checkout-api).

<CH.Scrollycoding>

## Enable the `useIndividualShippingEstimates` flag

> First, it is necessary to enable the [Checkout API](https://developers.vtex.com/docs/api-reference/checkout-api) behavior to fill the `shippingEstimateDate` with the SLA information. To do so, you must update the store's `orderForm` with the [Update orderForm configuration](https://developers.vtex.com/docs/api-reference/checkout-api#post-/api/checkout/pvt/configuration/orderForm) endpoint. You must include the flag `useIndividualShippingEstimates` to `true` as in the example shown.

Reviewer 2: "First, you need to enable the..."

"You must set the `useIndividualShippingEstimates` flag to `true` as in the example below.

```json Request mark=23
{
    "paymentConfiguration": {
    ...
    },
    "taxConfiguration": null,
    "minimumQuantityAccumulatedForItems": 1,
    "decimalDigitsPrecision": 2,
    "minimumValueAccumulated": null,
    "apps": [
    ...
    ],
    "allowMultipleDeliveries": true,
    "allowManualPrice": null,
    "savePersonalDataAsOptIn": false,
    "maxNumberOfWhiteLabelSellers": null,
    "recaptchaValidation": "vtexcriteria",
    "recaptchaMinScore": null,
    "recaptchaKeys": null,
    "maskStateOnAddress": true,
    "enableSecureCookies": true,
    "useOwnershipCookie": true,
    "ignoreProfileData": null,
    "useIndividualShippingEstimates": true
}
```

By enabling this flag, the `shippingEstimateDate` will be updated with the SLA information configured by the sellers.

---

> ## Add items to cart

Reviewer 2: "Adding items to the cart"

> Once you enable the `useIndividualShippingEstimates`, test if the configuration is correct by adding items to a shopping cart using the [Add cart items](https://developers.vtex.com/docs/api-reference/checkout-api#post-/api/checkout/pub/orderForm/-orderFormId-/items) endpoint. See the request body example presented.

Reviewer 2: "Once you enable the `useIndividualShippingEstimates` flag, test..."

"See the request body example below."

```json Request
{
  "orderItems": [
    {
      "seller": "1",
      "id": 6,
      "quantity": 1
    },
    {
      "seller": "1",
      "id": 11,
      "quantity": 1
    }
  ]
}
```

---

## Send shipping data attachment

> Send the item shipping data to the `orderForm` using the [Add shipping address and select delivery option](https://developers.vtex.com/docs/api-reference/checkout-api#post-/api/checkout/pub/orderForm/-orderFormId-/attachments/shippingData) endpoint as in the example shown.

Reviewer 2: "...as shown in the example."
